src/LibraryManager.Vsix/Contracts/Logger.cs

```diff
@@ -88,5 +90,142 @@ private static bool EnsurePane()
 
             return _pane != null;
         }
+
+        internal static void LogEventsHeader(OperationType operationType)
+        {
+            LogEvent(GetOperationHeaderString(operationType), LogLevel.Task);
+        }
+
+        internal static void LogEventsSummary(IEnumerable<ILibraryInstallationResult> totalResults, OperationType operationType, TimeSpan elapsedTime )
+        {
+            LogEvent(GetOperationHeaderString(operationType), LogLevel.Status);
+            LogOperationSummary(totalResults, operationType, elapsedTime);
+            LogEvent(string.Format(LibraryManager.Resources.Text.TimeElapsed, elapsedTime), LogLevel.Operation);
+            LogEvent(LibraryManager.Resources.Text.SummaryEndLine + Environment.NewLine, LogLevel.Operation);
+        }
+
+        private static void LogOperationSummary(IEnumerable<ILibraryInstallationResult> totalResults, OperationType operation, TimeSpan elapsedTime)
+        {
+            int totalResultsCounts = totalResults.Count();
+            IEnumerable<ILibraryInstallationResult> successfulRestores = totalResults.Where(r => r.Success);
+            IEnumerable<ILibraryInstallationResult> failedRestores = totalResults.Where(r => !r.Success);
+            IEnumerable<ILibraryInstallationResult> cancelledRestores = totalResults.Where(r => r.Cancelled);
+            IEnumerable<ILibraryInstallationResult> upToDateRestores = totalResults.Where(r => r.UpToDate);
+
+            bool allSuccess = successfulRestores.Count() == totalResultsCounts;
+            bool allFailed = failedRestores.Count() == totalResultsCounts;
+            bool allCancelled = cancelledRestores.Count() == totalResultsCounts;
+            bool allUpToDate = upToDateRestores.Count() == totalResultsCounts;
+            bool partialSuccess = successfulRestores.Count() < totalResultsCounts;
+
+            LogEvent(GetSummaryHeaderString(operation) + Environment.NewLine, LogLevel.Task); 
+            if (allUpToDate)
+            {
+                LogEvent(LibraryManager.Resources.Text.LibraryRestoredNoChange + Environment.NewLine, LogLevel.Operation);
+            }
+            else if (allSuccess)
+            {
+                string successText = operation == OperationType.Clean ?
+                                     string.Format(LibraryManager.Resources.Text.CleanLibrariesSucceeded, totalResultsCounts, Math.Round(elapsedTime.TotalSeconds, 2)) :
+                                     string.Format(LibraryManager.Resources.Text.LibrariesRestored, totalResultsCounts, Math.Round(elapsedTime.TotalSeconds, 2));
+
+                LogEvent(successText + Environment.NewLine, LogLevel.Operation);
+            }
+            else if (allCancelled)
+            {
+                string canceledText = operation == OperationType.Clean ?
+                                    string.Format(LibraryManager.Resources.Text.RestoreLibrariesCancelled, totalResultsCounts):
```

Comment:
This order feels backwards.  If it's Clean, you're using the Restore string?

Reply:
Please ignore these methods. I have re-written and wil be avialble in my next PR